openmetadata-ui/src/main/resources/ui/src/components/SearchSettings/TermBoost/TermBoost.tsx

```diff
@@ -0,0 +1,182 @@
+/*
+ *  Copyright 2025 Collate.
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+import Icon from '@ant-design/icons';
+import { Button, Col, Row, Slider, Typography } from 'antd';
+import { AxiosError } from 'axios';
+import React, { useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { ReactComponent as Delete } from '../../../assets/svg/delete-white.svg';
+import { TermBoost } from '../../../generated/configuration/searchSettings';
+import tagClassBase from '../../../utils/TagClassBase';
+import { showErrorToast } from '../../../utils/ToastUtils';
+import { AsyncSelect } from '../../common/AsyncSelect/AsyncSelect';
+import './term-boost.less';
+
+interface TermBoostProps {
+  termBoost: TermBoost;
+  onTermBoostChange: (termBoost: TermBoost) => void;
+  onDeleteBoost: (termValue: string) => void;
+}
+
+const TermBoostComponent: React.FC<TermBoostProps> = ({
+  termBoost,
+  onTermBoostChange,
+  onDeleteBoost,
+}) => {
+  const { t } = useTranslation();
+  const [formData, setFormData] = useState<TermBoost>({
+    field: termBoost.field ?? '',
+    value: termBoost.value ?? '',
+    boost: termBoost.boost ?? 0,
+  });
```

Comment:
Not sure why we want to manage formData and not using  https://4x.ant.design/components/form/#header